### DISCUSSION TITLE
QDialog not showing of focus

### DISCUSSION THREAD

**Discussion by LigiaAb** [quasarframework/quasar#13156] [General - Components / Directives / etc] [answered]
I´m trying to show a q-dialog but it doesnt show:

My code is something like:

In the component that have de input and the dialog
```html

<div v-for="input in data" v-bind:key="input.id"/>
...
<q-input ... @focus="input.visible = true" @input="(val)=>$emit(`${input.id}`, val)"... />
<q-dialog ... v-model="input.visible" ... > <!-- content --> </q-dialog>
...
<script>
...
props:{
   data:{
      require:true
   }
},
...
</script>
```

And in the page I'm using the component like this:

```html
...
<Form-component data="myInputArray" @input1="val=> input1 = val"/>
...
<script>
...
data:{
   return{
      input1: '',
      isVisible: false,
   }
},
computed:{
    myInputArray: {
       set: function (){
          reutrn  [{ id:'input1', value: this.input1, label: 'xxxx', visible: this.isVisible} ]
      }
       get: function (){
          reutrn  [{ id:'input1', value: this.input1, label: 'xxxx', visible: this.isVisible} ]
      }
   }
}
...
</script>
```

I already try to use the `myInputArray` var in `data` but change it latter to automatly update the variables in the object

Code example:
https://codesandbox.io/s/admiring-hopper-tjyein?file=/src/pages/Index.vue

**Comment by metalsadman** [comment, marked as answer]
it does show a dialog just empty probably caused by the warnings in your code, try fixing those first.

**Reply by LigiaAb** [reply]
After posting the question I continued experimenting on the link and managed to make it work, Thanks :)

I just added the visibility variable in `data` and add it in the set and get
